### PR TITLE
CNF-13528: ztp: reference: Use capturegroupInlineDiff for PTPConfig

### PR DESCRIPTION
**Pull request by lack**
Signed-off-by: Jim Ramsay <jramsay@redhat.com>

### CODE REVIEW

File: ztp/source-crs/PtpConfigBoundaryForEvent.yaml [hunk 1]
```diff
@@ -16,13 +16,13 @@ spec:
       logReduce: "true"
     ptp4lConf: |
       # The interface name is hardware-specific
-      [$iface_slave]
+      [(?<iface_slave>[[:alnum:]]+)]
```
Comment (lack) [inline ztp/source-crs/PtpConfigBoundaryForEvent.yaml:19]: @aneeshkp / @jzding / @josephdrichard : Does this look okay?  This changes the example from `$interface_slave` to an actual regex capturegroup `(?<interface_slave>...)` which maintains the idea of "Customers need to fill in this part", but adds the actual regex that the text should match.

It's a side-effect of the new facility we added in the reference, and if we really need to fixup the source-crs so they don't change, we can do that...

**Comment by lack** [comment]
Here is an example of what the cluster-compare tool v0.4.0 shows now against a somewhat contrived PtpConfigBoundary.yaml with differences:

```diff
Cluster CR: ptp.openshift.io/v1_PtpConfig_openshift-ptp_boundary
Reference File: optional/ptp-config/PtpConfigBoundary.yaml
Diff Output: diff -u -N /tmp/MERGED-1211397407/ptp-openshift-io-v1_ptpconfig_openshift-ptp_boundary /tmp/LIVE-3841780862/ptp-openshift-io-v1_ptpconfig_openshift-ptp_boundary
--- /tmp/MERGED-1211397407/ptp-openshift-io-v1_ptpconfig_openshift-ptp_boundary	2024-11-19 11:37:56.742487557 -0500
+++ /tmp/LIVE-3841780862/ptp-openshift-io-v1_ptpconfig_openshift-ptp_boundary	2024-11-19 11:37:56.742487557 -0500
@@ -5,7 +5,7 @@
   namespace: openshift-ptp
 spec:
   profile:
-  - name: boundary
+  - name: boundary-clock
     phc2sysOpts: -a -r -n 24
     ptp4lConf: |
       # The interface name is hardware-specific
@@ -30,6 +30,7 @@
       clockClass 248
       clockAccuracy 0xFE
       offsetScaledLogVariance 0xFFFF
+      Obviously a bad thing
       free_running 0
       freq_est_interval 1
       dscp_event 0
@@ -67,10 +68,10 @@
       unicast_req_duration 3600
       use_syslog 1
       verbose 0
-      summary_interval 0
+      summary_interval -4
       kernel_leap 1
       check_fup_sync 0
-      clock_class_threshold 135
+      clock_class_threshold 7
       #
       # Servo Options
       #
@@ -119,12 +120,14 @@
       userDescription ;
       timeSource 0xA0
     ptp4lOpts: "-2"
+    ptpClockThreshold:
+      holdOverTimeout: 600
+      maxOffsetThreshold: 100
+      minOffsetThreshold: -100
     ptpSchedulingPolicy: SCHED_FIFO
     ptpSchedulingPriority: 10
-    ptpSettings:
-      logReduce: "true"
   recommend:
   - match:
     - nodeLabel: node-role.kubernetes.io/master
     priority: 4
-    profile: boundary
+    profile: boundary-clock
```

File: ztp/kube-compare-reference/optional/ptp-config/PtpConfigDualCardGmWpc.yaml [hunk 2]
```diff
@@ -28,12 +28,12 @@ spec:
           MaxInSpecOffset: 100
         pins:
           {{- .plugins.e810.pins | toYaml | nindent 10 }}
-        #  "$iface_nic1":
+        #  "(?<iface_nic1>[[:alnum:]]+)":
```
Comment (aneeshkp) [inline ztp/kube-compare-reference/optional/ptp-config/PtpConfigDualCardGmWpc.yaml:31]: They are iface_timeTx all interface for GM are iface_timeTx

File: ztp/kube-compare-reference/optional/ptp-config/PtpConfigDualCardGmWpc.yaml [hunk 3]
```diff
@@ -109,32 +109,32 @@ spec:
...
       ts2phc.extts_correction 0
-      [$iface_nic2]
+      [(?<iface_nic2>[[:alnum:]]+)]
       ts2phc.master 0
       ts2phc.extts_polarity rising
       #this is a measured value in nanoseconds to compensate for SMA cable delay
       ts2phc.extts_correction -10
     ptp4lConf: |
-      [$iface_nic1]
+      [(?<iface_nic1>[[:alnum:]]+)]
```
Comment (aneeshkp) [inline ztp/kube-compare-reference/optional/ptp-config/PtpConfigDualCardGmWpc.yaml:123]: probably this should be  iface_nic_0 or iface_Tx0

File: ztp/source-crs/PtpConfigDualCardGmWpc.yaml [hunk 4]
```diff
@@ -13,7 +13,7 @@ spec:
   profile:
   - name: "grandmaster"
     ptp4lOpts: "-2 --summary_interval -4"
-    phc2sysOpts: -r -u 0 -m -w -N 8 -R 16 -s $iface_nic1 -n 24
+    phc2sysOpts: -r -u 0 -m -w -N 8 -R 16 -s (?<iface_nic1>[[:alnum:]]+) -n 24
```
Comment (aneeshkp) [inline ztp/source-crs/PtpConfigDualCardGmWpc.yaml:16]: iface_nic_0 or  iface_timeTx_0

**Review by aneeshkp — APPROVED** [review]
/lgtm

**Pull request merged**